traits/trait_types.py

```diff
@@ -3268,46 +3283,63 @@ class PrefixMap(TraitType):
 
     is_mapped = True
 
-    def __init__(self, map, **metadata):
+    def __init__(self, map, *, default_value=None, **metadata):
+        map = dict(map)
+        if not map:
+            raise ValueError("map must be nonempty")
         self.map = map
-        self._map = {}
-        for key in map.keys():
-            self._map[key] = key
 
-        try:
-            default_value = metadata.pop("default_value")
-        except KeyError:
-            if len(self.map) > 0:
-                default_value = next(iter(self.map))
-            else:
-                raise ValueError(
-                    "The dictionary of valid values can not be empty."
-                ) from None
+        if default_value is not None:
+            default_value = self._complete_value(default_value)
         else:
-            default_value = self.value_for(default_value)
+            default_value = next(iter(self.map))
 
         super().__init__(default_value, **metadata)
 
-    def value_for(self, value):
-        if not isinstance(value, str):
-            raise TraitError(
-                "Value must be {}, but a value {!r} was specified.".format(
-                    self.info(), value)
-            )
+    def _complete_value(self, value):
```

Comment:
It feels a little wrong to have this essentially duplicated in two classes, but two uses probably isn't enough to move it out as a utility function.

Reply:
Yep, that was pretty much exactly my thought, and I'm wavering - I'm happy to extract it you think it's worth it (but equally happy to leave it as-is).

Reply:
Looking at them they are different enough that they probably couldn't be pulled out separately anyway.